mbf_abstract_nav/src/planner_action.cpp

```diff
@@ -269,8 +269,8 @@ bool PlannerAction::transformPlanToGlobalFrame(
   for (iter = plan.begin(); iter != plan.end(); ++iter)
   {
     geometry_msgs::PoseStamped global_pose;
-    tf_success = mbf_utility::transformPose(robot_info_.getTransformListener(), robot_info_.getGlobalFrame(), iter->header.stamp,
-                                            robot_info_.getTfTimeout(), *iter, robot_info_.getGlobalFrame(), global_pose);
```

Comment:
the changes in this file seems unrelated to the initial problem. To me both timestamps would be fine. Not sure if users might have specific applications running against the old implementation

Reply:
Just adapt to the new, simpler interface of transformPose

Reply:
My point is that you change the behaviour from having a transform at request-timestamp to the newest transform. My concern is that this might brake existing user code. Since both implementations are reasonable, I'm inclined to keep the old one. @spuetz any preference here form your side?

Reply:
No, no;  we still get the newest transform. The only difference in behavior is that this transform will have the real time stamp, instead of time now. Actually, previous code was "lying"; with this PR we don't anymore

Reply:
Huh? I think you are talking about computeRobotPose/getRobotPose (which is a legit fix), and I'm talking transformPose and transformPoint: The old transformPose accepted target_time; the target_time was set to the time-stamp provided by the client (ex. line 272 in planner_action.cpp). Now transformPose uses the newest time.

Reply:
No, no; cause the new version uses the stamp provided with the pose; in this case, it's the same, as the pose is *iter and the stamp iter->header.stamp

Reply:
god you are right